src/common/Autocomplete/DesignationAccountAutocomplete.tsx

```diff
@@ -0,0 +1,70 @@
+import { FocusEventHandler, ReactElement } from 'react';
+import {
+  Autocomplete,
+  AutocompleteProps,
+  CircularProgress,
+  TextField,
+  TextFieldProps,
+} from '@mui/material';
+import { useGetDesignationAccountsQuery } from 'src/components/EditDonationModal/EditDonationModal.generated';
+
+interface DesignationAccountAutocompleteProps
+  extends Partial<AutocompleteProps<string, boolean, boolean, false>> {
+  accountListId: string;
+  onBlur?: FocusEventHandler<HTMLDivElement>;
+  textFieldProps?: Partial<TextFieldProps>;
+}
+
+export const DesignationAccountAutocomplete = ({
+  accountListId,
+  onBlur,
+  textFieldProps,
+  ...props
+}: DesignationAccountAutocompleteProps) => {
+  const {
+    data: designationAccountsData,
+    error: designationAccountsError,
+    loading,
+  } = useGetDesignationAccountsQuery({
+    variables: {
+      accountListId,
+    },
+  });
+
+  const designationAccounts =
+    designationAccountsData?.designationAccounts?.flatMap(
+      ({ designationAccounts }) => designationAccounts,
+    );
+
+  return (
+    <Autocomplete
+      autoSelect
+      autoHighlight
+      loading={loading}
+      {...props}
+      options={designationAccounts?.map(({ id }) => id) ?? []}
+      getOptionLabel={(accountId): string => {
+        const account = designationAccounts?.find(({ id }) => id === accountId);
+        return account ? `${account?.name} (${account.id})` : '';
+      }}
```

Comment:
You could do this to improve the code. I haven't tested it, so it might not work.

This will also mean you'll need to update the handleClick function.

```suggestion
      options={designationAccounts ?? []}
      getOptionLabel={(account): string => {
        return account ? `${account?.name} (${account.id})` : '';
      }}
```

Reply:
@dr-bizz This didn't really work. With some coercing on the type of account it probably could, but it is not super trivial from what we can tell.

Reply:
@kegrimes & @zweatshirt Lets jump on a call tomorrow

Reply:
@dr-bizz Curious what you did to get it to work